crates/doc/src/inference.rs

```diff
@@ -444,6 +445,82 @@ impl ObjShape {
             None
         }
     }
+
+    fn widen<'n, N>(&mut self, fields: &'n N::Fields, loc: Location, is_first_time: bool) -> bool
+    where
+        N: AsNode,
+    {
+        use crate::{Field, Fields};
+        // If a particular location defines `additionalProperties` as a subschema
+        // don't add the field from `AsNode`, instead jump straight to squashing
+        match self.additional.as_mut() {
+            // We only want to squash if your `additionalProperties`
+            // is set to something other than `false`.
+            Some(addl) if !addl.type_.eq(&types::INVALID) => {
+                fields.iter().enumerate().fold(false, |accum, (idx, node)| {
```

Comment:
nit: how about returning early here, ending the `match` block, so that what's currently the `_ =>` match arm can be de-indented 2x:

```
// If `additionalProperties` is anything other than `false`, then widen it with all `fields`.
if let Some(addl) = self.additional.as_mut() {
  if addl.type_ != types::INVALID {
    return fields.iter().enumerate().fold(false, |accum, (idx, node)| {
        accum || addl.widen(node.value(), loc.push_item(idx))
    })
  }
}

// Handle fields individually ...

```

Reply:
Fair enough. I had tried this originally and ran into https://github.com/rust-lang/rust/issues/53667 and thought the nested if statements were weird, but looking at this again it's still cleaner because of the early return. Changed 👍